core/db_classes/EE_Base_Class.class.php

```diff
@@ -732,1484 +631,1729 @@ protected function _prepare_datetime_field(
 
 
     /**
-     * This just takes care of clearing out the cached_properties
-     *
-     * @return void
-     */
-    protected function _clear_cached_properties()
-    {
-        $this->_cached_properties = [];
-    }
-
-
-    /**
-     * This just clears out ONE property if it exists in the cache
-     *
-     * @param string $property_name the property to remove if it exists (from the _cached_properties array)
-     * @return void
-     */
-    protected function _clear_cached_property($property_name)
-    {
-        if (isset($this->_cached_properties[ $property_name ])) {
-            unset($this->_cached_properties[ $property_name ]);
-        }
-    }
-
-
-    /**
-     * Ensures that this related thing is a model object.
-     *
-     * @param mixed  $object_or_id EE_base_Class/int/string either a related model object, or its ID
-     * @param string $model_name   name of the related thing, eg 'Attendee',
-     * @return EE_Base_Class
+     * @param $props_n_values
+     * @param $classname
+     * @return mixed bool|EE_Base_Class|EEM_CPT_Base
      * @throws ReflectionException
      * @throws InvalidArgumentException
      * @throws InvalidInterfaceException
      * @throws InvalidDataTypeException
      * @throws EE_Error
      */
-    protected function ensure_related_thing_is_model_obj($object_or_id, $model_name)
+    protected static function _get_object_from_entity_mapper($props_n_values, $classname)
     {
-        $other_model_instance = self::_get_model_instance_with_name(
-            self::_get_model_classname($model_name),
-            $this->_timezone
-        );
-        return $other_model_instance->ensure_is_obj($object_or_id);
+        // TODO: will not work for Term_Relationships because they have no PK!
+        $primary_id_ref = self::_get_primary_key_name($classname);
+        if (
+            array_key_exists($primary_id_ref, $props_n_values)
+            && ! empty($props_n_values[ $primary_id_ref ])
+        ) {
```

Comment:
There is a bunch of statements formatted like this in this file. Wouldn't the next be better ?

```suggestion
        if (array_key_exists($primary_id_ref, $props_n_values)
            && ! empty($props_n_values[ $primary_id_ref ])
        ) {
```

Or are we switching to a different PSR ?

Reply:
yes I would like to follow PSR12 now which replaces/extends PSR2

plz see: https://www.php-fig.org/psr/psr-12/#51-if-elseif-else

Reply:
Ok, understood. Need to change the PSR settings.